textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcherTest.java

```diff
@@ -0,0 +1,100 @@
+import edu.uci.ics.textdb.storage.LuceneDataStore;
+import edu.uci.ics.textdb.storage.reader.LuceneDataReader;
+import edu.uci.ics.textdb.storage.writer.LuceneDataWriter;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.w3c.dom.Attr;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Prakul
+ *
+ */
+
+
+public class KeywordMatcherTest {
+
+    private KeywordMatcher keywordMatcher;
+    private IDataWriter dataWriter;
+    private IDataReader dataReader;
+    private LuceneDataStore dataStore;
+    private IndexSearchSourceOperator indexSearchSourceOperator;
+
+    private Analyzer analyzer;
+    private Query queryObj;
+    private Schema schema;
+
+    @Before
+    public void setUp() throws Exception {
+        dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+        analyzer = new StandardAnalyzer();
+        dataWriter = new LuceneDataWriter(dataStore, analyzer);
+        dataWriter.clearData();
+        dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+        schema = dataStore.getSchema();
+    }
+
+    @After
+    public void cleanUp() throws Exception {
+        dataWriter.clearData();
+    }
+
+    public List<ITuple> getQueryResults(String query) throws DataFlowException, ParseException {
+        String defaultField = TestConstants.ATTRIBUTES_PEOPLE[5].getFieldName();
+
+        ArrayList<Attribute> attributeList = new ArrayList<Attribute>();
+        attributeList.add(TestConstants.ATTRIBUTES_PEOPLE[5]);
+        Analyzer analyzer = new StandardAnalyzer();
+        IPredicate predicate = new KeywordPredicate(query, attributeList, analyzer);
+        //ISourceOperator sourceOperator = new IndexSearchSourceOperator();
+
+        QueryParser queryParser = new QueryParser(defaultField, analyzer);
+        queryObj = queryParser.parse(query);
+        IDataReader dataReader = new LuceneDataReader(dataStore, queryObj);
+        indexSearchSourceOperator = new IndexSearchSourceOperator(dataReader);
+        //indexSearchSourceOperator.open();
+
+        keywordMatcher = new KeywordMatcher(predicate, indexSearchSourceOperator);
+        keywordMatcher.open();
+
+        List<ITuple> results = new ArrayList<ITuple>();
+        ITuple nextTuple = null;
+        while ((nextTuple = keywordMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+        indexSearchSourceOperator.close();
+        return results;
+    }
+
+
+    @Test
+    public void testKeywordMatcher() throws Exception {
+        String query = "short tall";
+        List<ITuple> list = getQueryResults(query);
```

Comment:
This test case is not enough.  We need to compare the returned results with expected results exactly.  Check the test cases in DictionaryMatcher as examples.  Added @rajesh9625 to this discussion.  In addition, you need to write more test cases.